### PR TITLE
add momentum correction in batchnorm

### PR DESCRIPTION
**Pull request by ailzhang**
This is used in distributed imagenet training on resnet. It is called every time when lr is updated. (ref: https://arxiv.org/abs/1706.02677)
Tested on resnet50 model together with pytorch/example/imagenet changes:
1 node, 8 gpus, batchsize=256, lr=0.1 (Baseline): * Prec@1 76.172 Prec@5 93.090
4 nodes, 32 gpus, batchsize=1024, lr=0.1(without the tricks): * Prec@1 74.980 Prec@5 92.144
4 nodes, 32 gpus, batchsize=1024, lr=0.4( with distributed training tricks): Prec@1 76.224 Prec@5 92.962

### CODE REVIEW

**Comment by apaszke** [comment]
Since "momentum correction" is not something standard yet, and `step` is meant to have the same interface across all optimizers, I'd rather not merge this PR. A different argument is that if someone uses multiple parameter groups, the correction might need to be different for each group, so this still doesn't work in this form. I think it's best to just make the buffer names we use in our optimizers consistent and part of the public interface, and make such things part of the training scripts. In case of imagenet example, one could just make the LR scheduling function decay momentum (mutate the optimizer) if it decays LR. We do that for LR already.

I'll close this for now, but feel free to continue the discussion here if you disagree.

**Comment by ailzhang** [comment]
Sure, that makes sense to me. I put momentum correction here since I also saw this term in other papers. I thought it would be good to add it into the interface. But you are right, we should handle it carefully to keep the interface consistent among all optimizers. So I'm good with not merging this :)
I moved this trick back to the example repo and updated the PR there. Please take a look when you have time. Thanks!
